salesforce/serializers.py

```diff
@@ -103,7 +104,8 @@ def create(self, validated_data):
             rd.contact_id = contact_id
             rd.save()
         except (ResourceDownload.DoesNotExist, IndexError):
-            rd = ResourceDownload.objects.create(**validated_data)
+            if book is not None:
```

Comment:
you should be able to just say `if book:` but not a big deal

Reply:
Fixed it.